src/assets/wise5/components/discussion/discussionController.ts

```diff
@@ -328,6 +329,16 @@ class DiscussionController extends ComponentController {
     }
   }
 
+  subscribeToAttachStudentAsset() {
+    this.subscriptions.add(
+      this.StudentAssetService.attachStudentAsset$.subscribe((data) => {
+        if (this.isForThisComponent(data)) {
+          this.attachStudentAsset(data.asset);
+        }
+      })
```

Comment:
Instead of a generic "data" param, what do you think about creating a new AttachStudentRequest interface?